pod/recorder/models.py

```diff
@@ -28,27 +30,88 @@
+            'Manager of this recorder. This manager will receive recorder '
+            'emails and he will be the owner of the published videos. If no '
+            'user is selected, this recorder will use manual assign system.'),
+        verbose_name=_('User'), null=True, blank=True)
+    # Default type of published videos by this recorder
+    type = models.ForeignKey(
+        Type, on_delete=models.CASCADE,
+        help_text=_('Video type by default.'))
+    # Directory name where videos of this recorder are published
+    directory = models.CharField(_('Publication directory'), max_length=50,
+                                 unique=True, help_text=_(
+            'Basic directory containing the videos published by the recorder.')
+                                 )
+
+    def __unicode__(self):
+        return "%s - %s" % (self.name, self.address_ip)
+
+    def __str__(self):
+        return "%s - %s" % (self.name, self.address_ip)
+
+    def ipunder(self):
+        return self.address_ip.replace(".", "_")
+
+    def save(self, *args, **kwargs):
+        super(Recorder, self).save(*args, **kwargs)
+
+    class Meta:
+        verbose_name = _("Recorder")
+        verbose_name_plural = _("Recorders")
+        ordering = ['name']
+
+
+class Recording(models.Model):
+    recorder = models.ForeignKey(Recorder,
+                                 on_delete=models.CASCADE,
+                                 verbose_name=_('Recorder'),
+                                 help_text=_('Recorder that made this '
+                                             'recording.'))
+    user = models.ForeignKey(User, on_delete=models.CASCADE,
+                             limit_choices_to={'is_staff': True},
+                             default=DEFAULT_RECORDER_USER_ID,
+                             help_text=_("User who has made the recording"))
     title = models.CharField(_('title'), max_length=200, unique=True)
     source_file = models.FilePathField(
-        path=DEFAULT_RECORDER_PATH, recursive=True)  # match="foo.*"
-    type = models.CharField(
-        max_length=50, choices=RECORDER_TYPE, default=RECORDER_TYPE[0][0])
+        path=DEFAULT_RECORDER_PATH, unique=True,
+        recursive=True)
+    type = models.CharField(max_length=50, choices=RECORDER_TYPE,
+                            default=RECORDER_TYPE[0][0])
     comment = models.TextField(_('Comment'), blank=True, default="")
-    date_added = models.DateTimeField(
-        'date added', default=timezone.now, editable=False)
+    date_added = models.DateTimeField('date added', default=timezone.now,
+                                      editable=False)
 
     def __str__(self):
-        return "%s" % (self.title)
+        return "%s" % self.title
 
     class Meta:
         verbose_name = _("Recording")
         verbose_name_plural = _("Recordings")
 
     def save(self, *args, **kwargs):
+        if self.source_file.endswith("zip"):
```

Comment:
I think we are going to have more than two types of recording ... If zip not match with audiovideocast... if no type, then video, I think it could do it

Reply:
please add comment to this function